client/lib/domains/check-inbound-transfer-status.js

```diff
@@ -1,17 +1,18 @@
 import wpcom from 'calypso/lib/wp';
 
-export function checkInboundTransferStatus( domainName, onComplete ) {
-	if ( ! domainName ) {
-		onComplete( null );
-		return;
-	}
+export function checkInboundTransferStatus( domainName ) {
+	return new Promise( ( resolve ) => {
+		if ( ! domainName ) {
+			resolve( { error: null } );
+		}
```

Comment:
This code is still more complicated than it needs to be. Nowhere in the affected code you want to use `new Promise()`. Here you can do:
```js
if ( ! domainName ) {
  return Promise.reject( new Error( 'missing domain parameter' ) );
}

return wpcom.req.get( ... );
```
Return the result of `wpcom.req.get` as it is. In case in success, the returned promise will resolve to the response body, in case of error it rejects with error.

Reply:
Excellent, thank you! I'd actually tried this approach on a previous attempt (inspired, actually, by the similar PR of yours that I reviewed last week). I ran into issues with the error states resolving though, and backtracked.

I see now the issue there wasn't actually anything to do with the promise or the `.catch` block... by returning this way and separating errors into `.catch`, I'd also separated them from from the `this.setState( { submittingWhois: false } );`, so when an error did happen, the page just never new it was supposed to stop waiting. I'm new enough to working with state that this slipped past me at first!